### ISSUE TITLE
Format fields replaced with dots after setting the same field of another sample to a longer string

### ISSUE THREAD

**Issue by yakupberk-sbg** [pysam-developers/pysam#554]
While creating a new VCF I found the error below.

```python
vcf2 = pysam.VariantFile('temp2.vcf', 'wb', header=vcf.header.copy())

rec1 = vcf2.new_record()
rec1.samples['TRUTH']['BLT'] = 'aaa'
rec1.samples['QUERY']['BLT'] = 'aaa'


rec2 = vcf2.new_record()
rec2.samples['TRUTH']['BLT'] = 'aaaa'
rec2.samples['QUERY']['BLT'] = 'aaa'


rec3 = vcf2.new_record()
rec3.samples['TRUTH']['BLT'] = 'aaaa'
rec3.samples['QUERY']['BLT'] = 'aaaaa'

rec4 = vcf2.new_record()
rec4.samples['QUERY']['BLT'] = 'aaa'
rec4.samples['TRUTH']['BLT'] = 'aaaa'


print rec1
print rec2
print rec3
print rec4
```

Output:
```
1	1	.	.	.	.	.	.	BLT	aaa	aaa

1	1	.	.	.	.	.	.	BLT	aaaa	aaa

1	1	.	.	.	.	.	.	BLT	....	aaaaa

1	1	.	.	.	.	.	.	BLT	aaaa	...
```

**Comment by bioinformed** [comment]
How is BLT defined?

**Comment by yakupberk-sbg** [comment]
The header line is below.
```
##FORMAT=<ID=BLT,Number=1,Type=String,Description="High-level location type (het|homref|hetalt|homalt|nocall).">
```

**Comment by cwhelan** [comment]
Is there any workaround for this issue? I am experiencing this behavior as well.

**Comment by jmarshall** [comment]
This was fixed by c0dbddbe03b936073df9060500550e7cbdba307a (PR #825) and is fixed in releases from 0.15.3 onwards.
